app/uk/gov/hmrc/hec/repos/HECTaxCheckStore.scala

```diff
@@ -66,76 +62,55 @@ trait HECTaxCheckStore {
 
 @Singleton
 class HECTaxCheckStoreImpl @Inject() (
-  mongo: ReactiveMongoComponent,
+  mongo: MongoComponent,
   configuration: Configuration
 )(implicit
   ec: ExecutionContext
-) extends HECTaxCheckStore
+) extends MongoCacheRepository(
+      mongoComponent = mongo,
+      replaceIndexes = true,
+      collectionName = "hecTaxChecks",
+      ttl = configuration.get[FiniteDuration]("hec-tax-check.ttl"),
+      timestampSupport = new CurrentTimestampSupport(), // Provide a different one for testing
+      cacheIdType = CacheIdType.SimpleCacheId // Here, CacheId to be represented with `String`
+    )
+    with HECTaxCheckStore
     with Logging {
 
   val key: String                      = "hec-tax-check"
   private val ggCredIdField: String    = s"data.$key.taxCheckData.applicantDetails.ggCredId"
   private val isExtractedField: String = s"data.$key.isExtracted"
 
-  private val hecIndexes = Seq(
-    Index(Seq("ggCredId" -> IndexType.Ascending)),
-    Index(
-      Seq("isExtracted" -> IndexType.Ascending),
-      partialFilter = Some(BSONDocument("isExtracted" -> false))
+  //indexes for hecTaxChecks collection
+  def mongoNewIndexes: Seq[IndexModel] = Seq(
```

Comment:
I would just call this `indexes` and also make it private

is there a reason it is a `def` instead of a `val`?

Reply:
there is variable indexes in the base class which is final. If you call it indexes , it will ask you to override it , but you can't because it's final :)
It has to be def , otherwise there is compile time error about uninitialised variable.

Reply:
ah ok, then you could call it something like `mongoIndexes` or whatever you think suits, just not sure the `new` bit is needed there.

will it work if you make it a `lazy val`?

Reply:
No, lazy val doesn't work. I have renamed it.